### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by kianmeng**
Found via `codespell -L te,wronly,noes,nulll` and `typos --format brief`

### CODE REVIEW

File: lib/pg/basic_type_registry.rb [hunk 1]
```diff
@@ -207,12 +207,12 @@ def register_type(format, name, encoder_class, decoder_class)
 
 	# Alias the +old+ type to the +new+ type.
 	def alias_type(format, new, old)
-		[:encoder, :decoder].each do |ende|
```
Comment (larskanis) [inline lib/pg/basic_type_registry.rb:210]: `ende` means encoder-decoder in this context. It is no spelling mistake.

**Review by larskanis — CHANGES_REQUESTED** [review]

**Comment by larskanis** [comment]
Thank you very much for scanning and correction! It's all OK with the exception of ende->end.

**Comment by larskanis** [comment]
Nice - thank you!

File: lib/pg/basic_type_registry.rb [hunk 2]
```diff
@@ -171,7 +171,7 @@ module Checker
 	include Checker
 
 	def initialize
-		# The key of these hashs maps to the `typname` column from the table pg_type.
+		# The key of these hashes maps to the `typname` column from the table pg_type.
```
Comment (cbandy) [inline lib/pg/basic_type_registry.rb:174]: Suggest: hash

File: ext/pg.c [hunk 3]
```diff
@@ -543,7 +543,7 @@ Init_pg_ext(void)
 	/* Result#result_error_field argument constant
 	 *
 	 * The SQLSTATE code for the error.
-	 * The SQLSTATE code identies the type of error that has occurred; it can be used by front-end applications to perform specific operations (such as error handling) in response to a particular database error.
+	 * The SQLSTATE code identities the type of error that has occurred; it can be used by front-end applications to perform specific operations (such as error handling) in response to a particular database error.
```
[the remark this answers is not in the document]
Comment (larskanis) [reply ext/pg.c:546]: Agreed.

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**